src/Compilers/CSharp/Portable/Lowering/SpillSequenceSpiller.cs

```diff
@@ -710,8 +710,15 @@ public override BoundNode VisitCatchBlock(BoundCatchBlock node)
             var locals = node.Locals;
 
             var exceptionFilterPrologueOpt = node.ExceptionFilterPrologueOpt;
-            Debug.Assert(exceptionFilterPrologueOpt is null); // it is introduced by this pass
             BoundSpillSequenceBuilder builder = null;
+            if (exceptionFilterPrologueOpt != null)
```

Comment:
I had trouble finding a test that fails if I remove this `if` block. Do we have one? (I think this requires that local rewriter results in spilling and runtime-async rewriter also results in spilling for a given `catch`)